lib/LedgerSMB/Template/DB.pm

```diff
@@ -131,13 +132,16 @@ does not call this without carefully whitelisting values.
 
 sub get_from_file {
     my ($package, $path, $language_code) = @_;
-    my $fname = $path;
+
     if ($path =~ m|/.*:| ){
        die 'Cannot run on NTFS alternate data stream!';
     }
-    $path =~ m|(.*)/([^/]+)$|;
-    $fname = $2;
+
+    my ( undef, undef, $fname)  = File::Spec->splitpath( $path);
+    die "Bad template file name $fname, should have one dot."
```

Comment:
The old code line 140  was looking for templatename.formatname, where neither can have a dot.   Now that is error checked.   

I am assuming a name like templatename.fmtname.moreStuffWithOrWithoutDots is not a well formed name here.

Your `split <regex>, <value>, 2` will yield 1 or 2 fields ( or list elements).  One field being the entire string if there is no match.  If there are matches, two fields will result.  The first  match divides the string into two fields.